evalml/pipelines/components/transformers/column_selectors.py

```diff
@@ -122,3 +123,44 @@ def transform(self, X, y=None):
             pd.DataFrame: Transformed X.
         """
         return super().transform(X, y)
+
+
+class SelectByTypeorTag(ColumnSelector):
+    """
+    Selects columns by specified Woodwork logical type or semantic tag in input data.
+
+    Arguments:
+        columns (list(string), list(ww.LogicalType)): List of datatypes, used to determine which columns to select.
```

Comment:
Don't we need an `__init__` method defined for this class in order to accept the types/tag(s) to use for the filter?

Reply:
This implementation just piggybacks off of the same argument name as defined in the parent class, so there's no need for an explicit `__init__`. That being said, it may make sense to rename this argument to differentiate between column names and column types. Thoughts?

Reply:
Got it, thanks for explaining.

I think the most pythonic thing to do here would be to add a second argument called "column_types" which is a list of types corresponding 1x1 positionally to the list of columns.

Overriding the type definition of an argument in the inheritance structure adds complexity. I don't think it'll cause problems in this particular case, since the inheritance structure is currently quite simple. But I do think using a separate argument would effectively future-proof this API against any unanticipated changes we may want to make to these components later, and is generally a useful pattern to follow for that reason.